[WIP] [TC-32] ATS Config File API

dg4prez opened the pull request:
Working API for ATS config file generation, divided by three "levels" - CDN, profile, and server.  The api supports three actions, fetch, db, and publish.

?action=fetch will check for a copy of the requested file on the filesystem, and if it does not exist will generate it from the DB.
?action=db will generate the file directly from the DB.
?action=publish will generate the file from the DB and write it to the filesystem.

No action defaults to fetch.

This is intended to be used for a redesign of the ATS config process.  No test files have been generated yet - this PR is intended for review only, and is linked to https://issues.apache.org/jira/browse/TC-32

mitchell852 commented:
looks like some of your formatting is off which makes it harder to read. i.e. curly brackets not lining up. did you run perltidy?

dg4prez commented:
Perltidy has been run.  I've added a route for ORT as well as handlers for the chkconfig and packages routes.  I've also added a "scope" parameter check for each file that is added to the ORT json.  We will need to add these parameters at installation time.

dg4prez commented:
Closing PR - a new one has been created for the PSQL rebase.
